src/BSeries.jl

```diff
@@ -1447,4 +1450,421 @@ include("latexify.jl")
     include("precompile.jl")
 end
 
+
+"""
+        is_energy_preserving(rka::RungeKuttaMethod,s)
+
+This function checks whether a Runge-Kutta method is energy 
+preserving for a given order s. 
+    Inputs: 
+        A: square matrix
+        b: vector of Butcher array
+        s: integer, the order
+    Output:
+        True/False
+"""
+function is_energy_preserving(rka::RungeKuttaMethod,p)
+#generate bseries 
+    series = bseries(rka, p)
+    #pass it to the main functoin 'is_energy_preserving'
+    is_energy_preserving(series)
+end
+
+"""
+
+        is_energy_preserving(series)::Bool
+This function checks whether the B-series `series` of a time integration
+method is energy preserving for Hamiltonian systems - up to the
+[`order`](@ref) of the `series`.
+References:
+This code is based on the Theorem 2 of the paper "Energy-Preserving 
+Integrators and the Structure of B-series".
+ (link: https://link.springer.com/article/10.1007/s10208-010-9073-1)
+ """
+function is_energy_preserving(series)
+    series_a = modified_equation(series)
+    #save all the coefficients in an array
+    coefficients = collect(values(series_a))
+    #save all the RootedTrees in another array: 
+    atrees = collect(keys(series_a))
```

Comment:
Please write comments that explain *why* you do something - *what* you do should become clear from the code itself.